Objects/codeobject.c

```diff
@@ -447,12 +469,17 @@ _PyCode_Update(struct _PyCodeConstructor *con, PyCodeObject *code)
         con->columntable = Py_None;
     }
 
-    Py_XDECREF(code->co_filename);
+    PyObject *newname = code->co_filename;
     Py_XDECREF(code->co_name);
     Py_XDECREF(code->co_qualname);
 
     init_code(code, con);
 
+    if (newname) {
```

Comment:
I wonder if newname will ever be NULL (I missed that the XDECREFs could probably be DECREFs in the last review).